### DISCUSSION TITLE
Vulnerabilities not shown in new releases

### DISCUSSION THREAD

**Discussion by AgustinRamiroDiaz** [artifacthub/hub#2441] [Q&A] [answered]
We have released new versions of https://artifacthub.io/packages/helm/okteto/okteto and the vulnerabilities aren't showing.

We've checked and didn't find any image not being private

**Comment by tegioz** [comment, marked as answer]
Hi @AgustinRamiroDiaz 👋 

It looks like no containers images have been detected since version `1.1.0-rc.1`. By default, Artifact Hub will try to extract the containers images used by Helm charts from the manifests generated from a dry-run install using the default values. This is done by using a [regular expression](https://github.com/artifacthub/hub/blob/ed8fd4a818ba191c5bf26f60f451df9e26e1218f/internal/tracker/source/helm/helm.go#L68-L70) that sometimes may not be able to identify them.

In cases where AH is not able to detect your containers images automatically, you can provide them manually with the [artifacthub.io/images annotation](https://artifacthub.io/docs/topics/annotations/helm/). As soon as a new version of your package with containers images is indexed, it'll be scanned for security vulnerabilities again 🙂

**Reply by AgustinRamiroDiaz** [reply]
I find it weird that we didn't change almost anything regarding the images. Also, testing the regex against the source files I find a lot of matches. 

Has there been any changes lately in this behavior? Is it expected that the regex could fail?

**Reply by tegioz** [reply]
Mmm no, [that part of the code hasn't changed in a long time](https://github.com/artifacthub/hub/blob/ed8fd4a818ba191c5bf26f60f451df9e26e1218f/internal/tracker/source/helm/helm.go#L548-L588).

I've run a quick test locally and it looks like the problem is that the Helm dry-run is failing on those versions (the regex is applied to the output of the dry-run, not to the sources directly). I got the error below when running it using the code linked above. You should be able to reproduce it with the Helm CLI tool. Versions < 1.1.0 have been processed fine, detecting the images as expected (all processed with the current version of the code).

Hope this helps!

```
2022/10/25 17:49:22 execution error at (okteto/templates/webhook-deployment.yaml:38:31): A valid .Values.subdomain is required
```

**Reply by AgustinRamiroDiaz** [reply]
Thanks tegioz! Very kind of you :bow:

**Reply by AgustinRamiroDiaz** [reply]
@tegioz One more question. That error is due to a validation we've purposely added to our chart to reduce the number of installations with wrong configuration.

What can we do to have the `validations` and the `ArtifactHub's CVE check` coexist?

**Reply by tegioz** [reply]
I think the easiest way would be to define the container images used in the annotation mentioned in my first comment.
